src/base/static/utils/expression/expression.d.ts

```diff
@@ -0,0 +1,33 @@
+import PropTypes from "prop-types";
```

Comment:
Why do we have this `.d.ts` file alongside other TS files?

>The "d.ts" file is used to provide typescript type information about an API that's written in JavaScript. 

https://stackoverflow.com/a/21247316/1884158

Reply:
Yeah, this might be the wrong use of a `.d.ts` file. I was thinking of the expression utility as a mini-API exposed to the rest of the app (similar to how ducks have their own `.d.ts` files).

In general I'm still trying to understand where type definitions should live in the codebase, especially if they need to be reused across different modules. I often think it would be nice to just make `types.tsx` files in folders where needed, but that doesn't seem to be a very common pattern, so I'm not sure.

Reply:
>In general I'm still trying to understand where type definitions should live in the codebase

Type definitions should live in the same file where the object that they are typing is implemented. But if that file is a js file, then we can create a `.d.ts` file alongside it. That's the only use case that we have for a `.d.ts` file.